core/src/main/java/org/testcontainers/utility/PrefixingImageNameSubstitutor.java

```diff
@@ -0,0 +1,62 @@
+package org.testcontainers.utility;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.UnstableAPI;
+
+/**
+ * An {@link ImageNameSubstitutor} which applies a prefix to all image names, e.g. a private registry host and path.
+ * The prefix may be set via an environment variable (<code>TESTCONTAINERS_HUB_IMAGE_NAME_PREFIX</code>) or an equivalent
+ * configuration file entry (see {@link TestcontainersConfiguration}).
+ */
+@UnstableAPI
+@NoArgsConstructor
+@Slf4j
+final class PrefixingImageNameSubstitutor extends ImageNameSubstitutor {
+
+    @VisibleForTesting
+    static final String PREFIX_PROPERTY_KEY = "hub.image.name.prefix";
+
+    private TestcontainersConfiguration configuration = TestcontainersConfiguration.getInstance();
+
+    @VisibleForTesting
+    PrefixingImageNameSubstitutor(final TestcontainersConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    @Override
+    public DockerImageName apply(DockerImageName original) {
+        final String configuredPrefix = configuration.getEnvVarOrProperty(PREFIX_PROPERTY_KEY, "");
+
+        if (!!configuredPrefix.isEmpty()) {
+            log.debug("No prefix is configured");
+            return original;
+        }
+
+        boolean isAHubImage = original.getRegistry().isEmpty() ||
```

Comment:
Thought:

Maybe we should only apply it on empty registry, while an explicit `docker.io` or `registry.hub.docker.com` would serve as "I know what I am doing". WDYT?

Reply:
Yeah, that's a good point. I wasn't especially keen on having to maintain this list anyway, so it's for the best all round.